rewrite-maven/src/main/java/org/openrewrite/maven/AddCommentToTag.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Incubating;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.marker.Markers;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Content;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.*;
+
+import static org.openrewrite.Tree.randomId;
+
+@Incubating(since = "7.24.0")
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class AddCommentToTag extends Recipe {
```

Comment:
I don't know if you intend to merge this recipe, but IMO it should better fit in rewrite-xml, with only the xpath as selector, no?

Reply:
> I don't know if you intend to merge this recipe, but IMO it should better fit in rewrite-xml, with only the xpath as selector, no?

An interesting idea! In this case, I don't think so, but I am open to suggestions. My intent for `@Incubating(since = "7.24.0")` is to communicate that the recipe will likely evolve over time based on feedback. I only spent a couple of minutes on this recipe, primarily to solicit feedback for issue [1802](https://github.com/openrewrite/rewrite/issues/1802). The issue is specifically asking for the ability to add comments in a tag for specific dependencies. So, I don't think `xPath` is enough.

For clarity, I'll rename the recipe to `AddCommentToMavenDependency`, and update the description and the display name.

@ferblaca has asked for support to add comments for various languages if/when it is possible. @murdos to your point, we could add a recipe `AddCommentToXmlTag` that only contains an `xPath`.

Reply:
@murdos I created an [issue](https://github.com/openrewrite/rewrite/issues/1817) to handle the request for XML.